src/worker/bpf.rs

```diff
@@ -0,0 +1,140 @@
+use std::{
+    cmp,
+    ffi::{c_char, CString},
+    fmt::Display,
+    mem, slice, thread, time,
+};
+
+use core_affinity::CoreId;
+use libc::{SYS_bpf, SYS_perf_event_open};
+use log::info;
+
+use aya_obj::copy_instructions;
+use aya_obj::generated::{
+    bpf_attach_type, bpf_attr, bpf_cmd, bpf_prog_type, perf_event_attr,
+    perf_event_sample_format, perf_type_id,
+};
+
+use crate::{BaseConfig, Worker, WorkerError, Workload, WorkloadConfig};
+
+#[derive(Debug, Clone, Copy)]
+pub struct BpfWorker {
+    config: BaseConfig,
+    workload: WorkloadConfig,
+}
+
+impl BpfWorker {
+    pub fn new(workload: WorkloadConfig, cpu: CoreId, process: usize) -> Self {
+        BpfWorker {
+            config: BaseConfig { cpu, process },
+            workload,
+        }
+    }
+}
+
+impl Worker for BpfWorker {
+    fn run_payload(&self) -> Result<(), WorkerError> {
+        info!("{self}");
+
+        let Workload::Bpf { nprogs, tracepoint } = self.workload.workload
+        else {
+            unreachable!()
+        };
+
+        for i in 0..nprogs {
```

Comment:
[nitpick] Could we move most of the content of the loop before it and simply change the name and reload it? I know it's silly, but there's a lot of code that seems to be running on every loop that doesn't need to.